test/test-parse-expression-statement.cpp

```diff
@@ -1369,6 +1369,30 @@ TEST(test_parse, let_as_statement_body_allows_asi) {
                 ElementsAre(spy_visitor::visited_variable_assignment{u8"x"}));
   }
 }
+
+TEST(test_parse, disallow_await_parameter_in_async_arrow_function) {
+  {
+    spy_visitor v;
+    padded_string code(u8"(async (await) => null)"_sv);
+    parser p(&code, &v);
+    p.parse_and_visit_expression(v);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",
+                                      "visit_enter_function_scope_body",
+                                      "visit_exit_function_scope"));
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_await_parameter_disallowed_in_async_arrow_function,
+                    await_token,
+                    offsets_matcher(&code, strlen(u8"(async ("), u8"await"))));
+  }
+
+  {
+    spy_visitor v = parse_and_visit_expression(u8"((await) => null)"_sv);
+    EXPECT_THAT(v.visits, ElementsAre("visit_enter_function_scope",
+                                      "visit_enter_function_scope_body",
+                                      "visit_exit_function_scope"));
+  }
```

Comment:
This is already tested by test_parse.variables_can_be_named_contextual_keywords, no?

Reply:
Good point